bindings.go

```diff
@@ -272,13 +251,17 @@ type Tree struct {
 	// Otherwise Parser may be GC'ed (and deleted by the finalizer) while some Tree objects are still in use.
 	p *Parser
 
+	input []byte
+
 	// most probably better save node.id
 	cache map[C.TSNode]*Node
 }
 
 // Copy returns a new copy of a tree
 func (t *Tree) Copy() *Tree {
-	return t.p.newTree(C.ts_tree_copy(t.c))
+	nt := t.p.newTree(C.ts_tree_copy(t.c))
+	nt.input = t.input
```

Comment:
does this need to use copy()?

Reply:
Ah, yeah it accidentally got reverted. I tried to update it to work with `Input` but it ended up getting too complicated.

Reply:
Hm, _do_ we actually need to copy it? Copying the tree makes sense because you can edit nodes; but as a user, I would expect either that I shouldn't ever change the underlying source array (in which case we don't need the copy), or that I can and it won't affect the tree (in which case we also need the copy when we first parse).

In other words, it feels inconsistent that the Tree creates its own, defensive copy of the input, but only if you call `Copy` — not on initial parse.

Personally, I think it makes sense to just always use the reference to the original.

Reply:
(My approval still stands, btw, above comment notwithstanding)

Reply:
I agree the discrepancy is weird. I'd rather not make a bunch of potentially expensive defensive copies so I'm inclined to agree with removing it as well.